watchtower/wtdb/client_db.go

```diff
@@ -719,24 +729,58 @@ func (c *ClientDB) CreateClientSession(session *ClientSession) error {
 			}
 		}
 
-		// Add the new entry to the towerID-to-SessionID index.
-		indexBkt := towerToSessionIndex.NestedReadWriteBucket(
-			towerID.Bytes(),
-		)
-		if indexBkt == nil {
-			return ErrTowerNotFound
+		// Get the session-ID index bucket.
+		dbIDIndex := tx.ReadWriteBucket(cSessionIDIndexBkt)
+		if dbIDIndex == nil {
+			return ErrUninitializedDB
+		}
+
+		// Get a new, unique, ID for this session from the session-ID
+		// index bucket.
+		nextSeq, err := dbIDIndex.NextSequence()
+		if err != nil {
+			return err
 		}
 
-		err = indexBkt.Put(session.ID[:], []byte{1})
+		// Add the new entry to the dbID-to-SessionID index.
+		newIndex, err := writeBigSize(nextSeq)
 		if err != nil {
 			return err
 		}
 
+		err = dbIDIndex.Put(newIndex, session.ID[:])
+		if err != nil {
+			return err
+		}
+
+		// Also add the db-assigned-id to the session bucket under the
+		// cSessionDBID key.
 		sessionBkt, err := sessions.CreateBucket(session.ID[:])
 		if err != nil {
 			return err
 		}
 
+		err = sessionBkt.Put(cSessionDBID, newIndex)
+		if err != nil {
+			return err
+		}
+
+		// TODO(elle): migrate the towerID-to-SessionID to use the
```

Comment:
yes :) I dont think the storage improvements are (at least at the moment) significant enough to do the migration in this PR. But could be worth it later on. So just left the todo so that it is clear that we could possibly get some storage gains here if needed.

Reply:
sgtm